addons/account/models/account_move.py

```diff
@@ -3020,10 +3020,11 @@ def _onchange_product_id(self):
             # E.g. mapping a 10% price-included tax to a 20% price-included tax for a price_unit of 110 should preserve
             # 100 as balance but set 120 as price_unit.
             if line.tax_ids and line.move_id.fiscal_position_id:
-                line.price_unit = line._get_price_total_and_subtotal()['price_subtotal']
+                quantity = line.quantity or 1
+                line.price_unit = line._get_price_total_and_subtotal()['price_subtotal'] / quantity
                 line.tax_ids = line.move_id.fiscal_position_id.map_tax(line.tax_ids._origin, partner=line.move_id.partner_id)
                 accounting_vals = line._get_fields_onchange_subtotal(price_subtotal=line.price_unit, currency=line.move_id.company_currency_id)
-                balance = accounting_vals['debit'] - accounting_vals['credit']
+                balance = (accounting_vals['debit'] - accounting_vals['credit']) * quantity
```

Comment:
To avoid multiple useless assignations to `price_unit`, I suggest:
```
if line.tax_ids and line.move_id.fiscal_position_id:
    price_subtotal = line._get_price_total_and_subtotal()['price_subtotal']
    line.tax_ids = line.move_id.fiscal_position_id.map_tax(line.tax_ids._origin, partner=line.move_id.partner_id)
    accounting_vals = line._get_fields_onchange_subtotal(price_subtotal=price_subtotal, currency=line.move_id.company_currency_id)
    balance = accounting_vals['debit'] - accounting_vals['credit']
    business_vals = line._get_fields_onchange_balance(balance=balance)
    if 'price_unit' in business_vals:
        line.price_unit = business_vals['price_unit']

```

Reply:
Ok tested it and tried to understand further what it is all about. Besides it works now for my use case too.
I changed the commit message and pushed it again. Thank you.